src/main/java/io/cdap/plugin/ariba/source/AribaStructuredTransformer.java

```diff
@@ -0,0 +1,366 @@
+    List<Schema.Field> cdapFields = outputSchema.getFields();
+
+    cdapFields.forEach(field -> {
+      String fieldName = field.getName();
+      Object fieldValStr = getFieldNativeValue(nativeRawRecord, fieldName);
+      Schema childSchema = field.getSchema();
+
+//       eg.
+//        {
+//          "Owner": {
+//            "SourceSystem": "ASM",
+//            "UserId": "customersupportadmin",
+//            "PasswordAdapter": "PasswordAdapter"
+//            }
+//        }
+      if (nativeRawRecord.get(fieldName) != null &&
+        nativeRawRecord.get(fieldName).getNodeType().name()
+          .equalsIgnoreCase(ResourceConstants.OBJECT) && !childSchema.getType().isSimpleType()) {
+        fieldValStr = readFields((JsonNode) fieldValStr, childSchema.getUnionSchema(0));
+      }
+
+//       e.g.
+//       {
+//         "Suppliers":[{
+//            "Suppliers": {
+//              "SourceSystem": "ASM",
+//              "SupplierId": "",
+//              "SupplierLocationId": ""
+//              }
+//           }]
+//       }
+      if (nativeRawRecord.get(fieldName) != null &&
+        nativeRawRecord.get(fieldName).getNodeType().name()
+          .equalsIgnoreCase(ResourceConstants.ARRAY) && !childSchema.getType().isSimpleType()) {
+        fieldValStr = readInternalDeltaFeed(childSchema, (JsonNode) fieldValStr);
+      }
+
+//         "State": "Active",
+      if (fieldValStr != null) {
+        processSchemaTypeValue(childSchema, recBuilder, fieldName, fieldValStr);
+      }
+    });
+    return recBuilder.build();
+  }
+
+  private List<StructuredRecord> readInternalDeltaFeed(Schema recordSchema, JsonNode fieldValStr) {
+    ObjectMapper mapper = new ObjectMapper();
+    if (fieldValStr.isArray() && recordSchema.getNonNullable().getComponentSchema() != null) {
+      try {
+        List<JsonNode> jsonList = mapper.readValue(fieldValStr.toString(), new TypeReference<List<JsonNode>>() {
+        });
+        return jsonList.stream().map(
+          field -> readFields(field.get(recordSchema.getNonNullable().getComponentSchema().getRecordName()),
+                              recordSchema.getNonNullable().getComponentSchema())).collect(Collectors.toList());
+      } catch (JsonProcessingException e) {
+        LOG.error("Error in Processing Structured Transformer from Json, cause: {}", e.getMessage());
+      }
+    }
+    return Collections.emptyList();
+  }
+
+
+  private void processSchemaTypeValue(Schema fieldSchema, StructuredRecord.Builder recordBuilder,
+                                      String fieldName, Object fieldValue) {
+
+    // Get Non-nullable schema object for the current field
+    Schema nonNullSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
+    if (fieldValue instanceof StructuredRecord || fieldValue instanceof List) {
+      recordBuilder.set(fieldName, fieldValue);
+    } else {
+      boolean isBlank = handleBlankVal(recordBuilder, fieldName, (String) fieldValue,
```

Comment:
Cast to string seems incorrect because fieldValue might not be a string

Reply:
Fixed.
Added check similar to above code for checking "instanceOf String".